tsdb/index/tsi1/index.go

```diff
@@ -147,35 +147,36 @@ func (i *Index) Open() error {
 	}
 
 	// Open all the Partitions in parallel.
+	partitionN := len(i.partitions)
 	n := i.availableThreads()
 
 	// Store results.
-	errC := make(chan error, n)
+	errC := make(chan error, partitionN)
 
 	// Run fn on each partition using a fixed number of goroutines.
 	var pidx uint32 // Index of maximum Partition being worked on.
 	for k := 0; k < n; k++ {
-		go func() {
+		go func(k int) {
 			for {
 				idx := int(atomic.AddUint32(&pidx, 1) - 1) // Get next partition to work on.
-				if idx >= len(i.partitions) {
+				if idx >= partitionN {
 					return // No more work.
 				}
 				errC <- i.partitions[idx].Open()
 			}
-		}()
+		}(k)
 	}
 
 	// Check for error
-	for i := 0; i < cap(errC); i++ {
+	for i := 0; i < partitionN; i++ {
 		if err := <-errC; err != nil {
 			return err
 		}
 	}
```

Comment:
This error check would continue on after receiving `threadN` responses instead of `partitionN` responses. The channel has a capacity of `threadN` so once this continues on then the goroutines above get blocked from adding more error responses.

This issue may have only been noticeable on computers with a smaller number of cores since as `threadN` gets closer to `partitionN` then it's less likely to occur. I have 4 threads on my machine and 16 partitions so it happened on every startup.

Reply:
@benbjohnson I fixed this issue in the other methods. Got about a 5% bump in ingestion. I didn't update `Open` so your PR should be able to merge on top.